src/main/java/com/google/sps/servlets/PotentialMatchesServlet.java

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.common.collect.ImmutableSet;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.friend_map.UserNode;
+import com.google.sps.data.friend_map.UserFriendsMap;
+import com.google.sps.data.PotentialMatchAlgorithm;
+
+/**
+*  Handles requests for getting the next potential match for a user's feed page.
+*/
+@WebServlet("/potential-matches")
+public class PotentialMatchesServlet extends HttpServlet {
+  private static final String USER_ENTITY = "User";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String NO_POTENTIAL_MATCH_RESULT = "NO_POTENTIAL_MATCHES";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String currUserID = request.getParameter("userid");
+    
+    String nextPotentialMatchID = loadUserPotentialMatch(currUserID);
+
+    response.setContentType("text/html");
+    response.getWriter().print(nextPotentialMatchID);
```

Comment:
The text returned here would be the next potential match id, but it does seem like a better idea to have this as a JSON object in case we need to later add in more data.